File: src/main/java/com/clinica/atendimento/service/AbstractServiceHandler.java
package com.clinica.atendimento.service;

import lombok.Getter;
import org.springframework.web.socket.WebSocketSession;

import java.io.IOException;

@Getter
public abstract class AbstractServiceHandler<T> {

    private final Class<T> type;

    public AbstractServiceHandler(Class<T> clazz) {
        this.type = clazz;
    }

    public abstract void process(WebSocketSession session, T t) throws IOException;

}

File: src/main/kotlin/com/clinica/atendimento/service/AbstractServiceHandler.kt
package com.clinica.atendimento.service

import org.springframework.web.socket.WebSocketSession
import java.io.IOException

abstract class AbstractServiceHandler<T>(
    val type: Class<T>
) {
    @Throws(IOException::class)
    abstract fun process(session: WebSocketSession, t: T)
}